### ISSUE TITLE
CI Failure: InputFiles Burgers Step Yaml Execute SegFault

### ISSUE THREAD

**Issue by pajkosmi** [sxs-collaboration/spectre#5384]
# Bug reports:

While waiting for CI to pass on a local, recently rebased branch, only on **Unit Tests (gcc-9, Debug)** does the "Test formaline tar can be built" step fail.

### Expected behavior:

CI completes successfully.

### Current behavior:

Only on **Unit Tests (gcc-9, Debug)**, during the "Test formaline tar can be built" step, `2 - InputFiles.Burgers.Step.yaml.execute (Failed)` occurs with a Segmentation Fault.  The entire error message is posted below under "Detailed Discussion".

### Environment:

This occurs on CI.

# Feature request:

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Desired feature:

- [ ] Detail 1
- [ ] Detail 2
- [ ] Detail 3

### Detailed discussion:

Error file below:
[burgers_error.txt](https://github.com/sxs-collaboration/spectre/files/12432259/burgers_error.txt)

I don't know if they're related, but this same test failed (among others) mentioned in issue #1031, specific to the Travis system.

**Comment by nilsvu** [comment]
I don't immediately see what might be causing this. I've seen it on other CI builds as well (clang-10 Debug most recently), also in the "formaline" test. AFAICT the formaline build configuration is the same as the full build..

**Comment by nilsdeppe** [comment]
Did this start when we switched to formaline sharing ccache with the first build?

**Comment by nilsvu** [comment]
No it happened before :(

**Comment by nilsvu** [comment]
Here it happened while running input file tests (not in formaline): https://github.com/sxs-collaboration/spectre/actions/runs/6193508313/job/16815188173#step:20:1
Also in Burgers though...

**Comment by nilsvu** [comment]
This is happening a lot. @wthrowe any idea what might be going on? It seems to only affect the Burgers executable. Recent failure: https://github.com/sxs-collaboration/spectre/actions/runs/6489244786/job/17623141811?pr=5546

**Comment by wthrowe** [comment]
No.  I've seen it in CI a few times, but don't recall ever seeing it locally, although I don't use older gcc builds or builds of formaline.  From googling apparently the "constprop" in the backtrace means there's some sort of inlining going on, but that doesn't mean it's relevant.  The  original report doesn't have that frame, but it might have just inlined more for some reason.
